doc/secure_upgrade/secure_upgrade.md

```diff
@@ -0,0 +1,179 @@
+- We will call a dedicated script to create an image signature:
+We will allow the use of 2 script types in this flow -
+
+	a.  development script - sign_image_dev.sh, which will use simple cms OpenSSL signing (enable with flag set to SECURE_UPGRADE_MODE == 'dev') with development keys (path provided in SECURE_UPGRADE_DEV_SIGNING_KEY and SECURE_UPGRADE_DEV_SIGNING_CERT).
+
+	b.  production script - sign_image_${platform}.sh (enable with the flag set to SECURE_UPGRADE_MODE == 'prod') that each vendor will be able to provide for themselves. When implementing this script, it is required to implement the method sign_image_prod(){} which is being called with arguments $output_image and $out_signature during the build process and is supposed to create a signature in $out_signature that will be concatenated to the image in the calling build script, afterward.
+
+![signing_options](signing_options.png)
+
+####  1.7.2. <a name='Verificationprocess'></a>Verification process
+- SONIC install – changes will be mainly done in sonic-installer main.py where bootloader methods are being called and will only affect 'sonic-installer install' command (other sonic-installer commands will not be affected). The installation process is triggered by install.sh script, which is different for every platform.
+In our opinion, python modifications will be easier to add and maintain than install.sh changes, as changing install.sh (for each platform) will require much more sources to maintain and test (different HW to check on, test cases for each HW, etc.).
+- A signed image will go through a verification process before installation, as main.py will call a dedicated bash script to split the signature from the image and check it with a public key certificate.
+Verification script called verify_image_sign.sh will verify using OpenSSL cms. Verification is enabled only if efi tools are enabled, and "Secure Boot" flag is enabled in BOIS. Accordingly, the certificate will be fetched from BIOS using EFI tools.
+In order to see if secure boot flag is enabled, user can use the linux command 'bootctl status' and look for "Secure Boot: enabled/disabled" line, or use the command:
+
+        bootctl status 2>/dev/null | grep "Secure Boot:"
+
+![dev_verification_script](dev_verification_script.png)
+
+###  1.8. <a name='SAIAPI'></a>SAI API 
+
+NA
+
+###  1.9. <a name='Configurationandmanagement'></a>Configuration and management 
+
+NA
+
+####  1.9.1. <a name='CLIYANGmodelEnhancements'></a>CLI/YANG model Enhancements 
+
+NA
+
+####  1.9.2. <a name='ConfigDBEnhancements'></a>Config DB Enhancements  
+
+NA
+		
+###  1.10. <a name='WarmbootandFastbootDesignImpact'></a>Warmboot and Fastboot Design Impact  
+NA
+
+###  1.11. <a name='RestrictionsLimitations'></a> Restrictions/Limitations  
+####  1.11.1. <a name='uefitoolsandsecurebootenabled'></a> uefi tools and secure boot enabled
+Device arch should support secure boot including UEFI tools.
```

Comment:
what are the UEFI tools being used to support secure upgrade process?

Reply:
added the list and url